brave/src/test/java/brave/features/baggage/SingleHeaderCodec.java

```diff
@@ -77,7 +46,9 @@ public BaggageCodec build() {
   }
 
   @Override public boolean decode(ExtraBaggageFields extra, Object request, String value) {
-    assert extra.isDynamic() : "This is unlikely to work with fixed fields!";
+    if (!extra.isDynamic()) {
```

Comment:
It is important to understand that with or without dynamic fields, something like a consolidated header is possible. All "dynamic does" is allow you to use fields not whitelisted in BaggagePropagation setup. Encode/decode is decoupled from these details now, especially in a follow-up change that will remove the "extra" parameter from decode.